readme.md

```diff
@@ -45,7 +45,7 @@ Renovate works on these platforms:
 
 Renovate is widely used in the developer community:
 
-![Renovate Matrix](./docs/usage/assets/images/matrix.png)
+![Logos of companies that use Renovate: Prisma, Netlify, Envoy, Condé Nast, Microsoft, Atlassian, Sourcegraph, Mozilla, Deloitte, Telus, Yarn, HashiCorp, Automattic, Algolia, eBay, Cypress, Red Hat, Financial Times, Uber, Buildkite](./docs/usage/assets/images/matrix.png)
```

Comment:
A long alt text means the screenreader will keep talking at you about the image.

A plain text list works really well with a screen reader, as it understands the list semantics, and notifies you when you begin and end the list. And you can step through the items in the list, and go back and forth. 🙂

Moving the list into plain text means we can keep the alt text on the image _short_: "Logos of companies that use Renovate".

Reply:
Can you mock up how that would look? I prefer the alt text for now, but open to it

Reply:
# Mock up

## Who Uses Renovate?

Renovate is used by these companies or projects:

- Prisma
- Netlify
- Envoy
- Condé Nast
- Microsoft
- Atlassian
- Sourcegraph
- Mozilla
- Deloitte
- Telus
- Yarn
- HashiCorp
- Automattic
- Algolia
- eBay
- Cypress
- Red Hat
- Financial Times
- Uber
- Buildkite

![Logos of companies or projects that use Renovate](./docs/usage/assets/images/matrix.png)

## Renovate OSS Insights

...

# Code of mockup

```markdown

## Who Uses Renovate?

Renovate is used by these companies or projects:

- Prisma
- Netlify
- Envoy
- Condé Nast
- Microsoft
- Atlassian
- Sourcegraph
- Mozilla
- Deloitte
- Telus
- Yarn
- HashiCorp
- Automattic
- Algolia
- eBay
- Cypress
- Red Hat
- Financial Times
- Uber
- Buildkite

![Logos of companies or projects that use Renovate](./docs/usage/assets/images/matrix.png)

## Renovate OSS Insights

...
```

Reply:
Sorry, but I don't like it being so long. It's meant to be a quick flash of color and validity and not take up a page of scrolling. Maybe it's not necessary to list all of them in the alt text and can say something like "Logos of 20 companies which use Renovate, including Mozilla, Deloitte and Red Hat"?

Reply:
## Listing twenty companies is too much

> Sorry, but I don't like it being so long. It's meant to be a quick flash of color and validity and not take up a page of scrolling.

I agree, the list is too long. :smile: 

## Image alt should be equivalent to image

> Maybe it's not necessary to list all of them in the alt text and can say something like "Logos of 20 companies which use Renovate, including Mozilla, Deloitte and Red Hat"?

Hmm. I read that generally alt text is supposed to offer a meaningful equivalent to the image. If all the companies in the image are important, than the alt text should list all companies as well. But then you end up with a wall of speech. :disappointed: 

## Table of possible solutions

Note that we show 20 companies in the current image.

| My rank | Companies in image | Plain text list in readme | Alt image text |
| --- | --- | --- | --- |
| 1 | about 5 really big fishes | No | Short, lists 5 big fishes |
| 2 | 10 | Yes | Short, generic |
| 3 | 20 | Yes, but _hidden_ within expandable `<details>` element, need to check if this is properly accessible first! | Short, generic |
| 4 | 20 | No | Only describe important companies, but break equivalency |
| 5 | 20 | No | Full description of all companies |

Reply:
Saying that the alt text has to list every company is saying that the alt text of any image must list everything in the image (e.g. if you describe one thing, you have to describe everything, and no "etc." allowed). I'm not an expert, but I think it's a bad limitation/restriction if so

Reply:
Hmm, this is a tricky subject. It seems that _context is everything_, as the [WebAIM site, alt text instructions](https://webaim.org/techniques/alttext/) says.

## Quotes from WebAIM's instructions for alternative text

These quotes seem most relevant to me, for now. But read the whole document to learn more. :wink: 

### Provide content and meaning in alt text

> Although technology is getting better at recognizing what an image **depicts**, algorithms alone cannot understand what an image **means** within the context of the overall page. A maple leaf might represent Canada, or it might just illustrate the leaf of a tree. Web page authors must provide alternative text that represents the **content** and **function** of their images.

### All images must have an `alt` attribute

> **Every** image should have an `alt` attribute, even if it's `alt=""` (sometimes called "null" alternative text).

### Decorative images should have `alt=""`

> A "decorative" image is one that...
> 
> - does not present important content,
> - is used for layout or non-informative purposes, and
> - does not have a function (e.g., is not a link).
> 
> Decorative images should have `alt=""`.

### Important rules

> The `alt` attribute should typically:
> 
> - **be accurate and equivalent** in representing content and function.
> - **be succinct.** Content (if any) and function (if any) should be presented as succinctly as possible, without sacrificing accuracy. Typically, only a few words are necessary, though rarely a short sentence or two may be appropriate.
> - **not be redundant** or provide the same information as text near the image.
>  - **not include phrases like "image of ..." or "graphic of ...", etc.** This would be redundant since screen readers already announce "graphic" along with the alt text. If the fact that an image is a photograph or illustration, etc. is important content, it may be useful to include this in alternative text.

### Logos

> Many websites link the main branding logo to the home page. Providing alternative text for the image, such as your company name (`alt="Acme Company"`), will usually suffice. The word "logo" (`alt="Acme Company Logo"`) is usually not an important part of the image's content or function. Similarly, it is usually not necessary to indicate that the image links to the home page (`alt="Acme Company home page"`) because this is a common convention—hearing "Link, Graphic, Acme Company" at the top of a web page is sufficient for a screen reader user to know that it is a logo linked to the home page.

Reply:
This was a triumph! I'm making a note here, huge success. It's hard to overstate my satisfaction.

Turns out I can use a _collapsible_ list beneath the "twenty logos image". iOS's VoiceOver works perfect with that. :partying_face: 

Turns out the cake was not a lie, and we can have our cake and eat it too:

- keep big image of 20 companies
- short alt text for image
- collapsible list (so it's out of the way for seeing users)
- plain text list after expanding for screenreader users